### PR TITLE
New features for data viewer

### PR DESCRIPTION
**Pull request by Dtphelan1**
Includes pagination, an all precincts view, flter on rep name, better share button placement, resolved render errors.

### CODE REVIEW

File: front-end/src/components/TableInfoFilters.js [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+          onChange={onTMMChange}
+          id="tmmSearch"
+          placeholder={copy.tmmPlaceholder}
+        />
+      </div>
+      {/* Article Select */}
+      <div className="filter">
+        <label>{copy.articleFilterLabel}</label>
+        {useMemo(
+          () => (
```
Comment (GrahamGoudeau) [inline front-end/src/components/TableInfoFilters.js]: I don't think I've seen a hook used here in a JS expression mid-template before. What is this used for? Just for my own knowledge

File: front-end/src/components/VotingHistoryTable.js [hunk 2]
```diff
@@ -11,11 +11,12 @@ function repUpForReelection(representative) {
   return representative.reelection === '2021';
 }
 
-function Table({ data, articleFilters, reelectionToggle, articles }) {
+function Table({ data, precinct, articleFilters, tmm = '', reelectionToggle, articles }) {
+  console.log('precinct', precinct);
```
Comment (GrahamGoudeau) [inline front-end/src/components/VotingHistoryTable.js]: remove console

File: front-end/src/components/VotingHistoryTable.js [hunk 3]
```diff
@@ -11,11 +11,12 @@ function repUpForReelection(representative) {
...
   const tableData = useMemo(() => {
     const mappedData = [];
     data.forEach(({ representative, votes }) => {
-      const voteObj = { member: representative.fullName };
+      const voteObj = { member: representative.fullName, precinct: representative.precinct, dummycol: 'ayo' };
```
Comment (GrahamGoudeau) [inline front-end/src/components/VotingHistoryTable.js]: Should this get removed/cleaned up?

File: front-end/src/components/VotingHistoryTable.js [hunk 4]
```diff
@@ -124,8 +159,32 @@ function Table({ data, articleFilters, reelectionToggle, articles }) {
           })}
         </tbody>
       </table>
+      <div id="pagination-controls" className="d-flex justify-content-center align-items-center ml-0 mt-3 mb-3 mr-0">
+        <button
```
Comment (GrahamGoudeau) [inline front-end/src/components/VotingHistoryTable.js]: I think two non-critical suggestions for this pagination:

1. Should we hide it entirely when there's just a single page? ie when no pagination is possible?
2. On mobile, should we put the pagination above the table when we display it?

File: front-end/src/components/VotingHistoryTable.js (hunk 2 above)
Comment (Dtphelan1) [reply front-end/src/components/VotingHistoryTable.js]: Never.

File: front-end/src/components/VotingHistoryTable.js (hunk 3 above)
Comment (Dtphelan1) [reply front-end/src/components/VotingHistoryTable.js]: Absolutely not thank you for asking

File: front-end/src/components/VotingHistoryTable.js (hunk 4 above)
Comment (Dtphelan1) [reply front-end/src/components/VotingHistoryTable.js]: 1) I would be happy to try rendering this conditionally when pagination is needed. 
2) I'm not sure I have strong intuitions here. What's driving your thought process? I'd be willing to explore it, but don't know of what would be preferable one way or another honestly.

File: front-end/src/components/TableInfoFilters.js (hunk 1 above)
Comment (Dtphelan1) [reply front-end/src/components/TableInfoFilters.js]: Yeah so the useMemo hook tells a component when it needs to re-render. I was seeing a bunch of unnecessary re-renders of the select components when other filter-values they didn't care about were being updated. If we were _really_ good devs, we'd break these out into separate components and do the memoization before export, but I think this is sufficient for now. Hopefully? 😄

**Review by GrahamGoudeau — APPROVED** [review]

**Review by mgramigna — APPROVED** [review]
Looks good. Great improvements

File: front-end/src/components/VotingHistoryTable.js (hunk 4 above)
Comment (Dtphelan1) [reply front-end/src/components/VotingHistoryTable.js]: @GrahamGoudeau bump

**Comment by Dtphelan1** [comment]
Hello Gamers. 

This is a friendly heads up that the PR has changed significantly since last review. I finally got around to a collapse-on-mobile UI for the table-filters, made some small improvements to the footer, and more modifications to the table itself (smaller font; changed padding on the info-button). 

Feel free to take a look. Waiting until merge to deploy a new version, so check on localhost ;) 

Thanks for listening, and I cannot wait to hop off the battle bus with you all in Fortnite. For now, I need to get back to playing Animal Crossing New Leaf on my Nintendo DS.

**Comment by Dtphelan1** [comment]
bois slow down with the reviews plz

**Comment by Dtphelan1** [comment]
Deployed a new version of this if you want to see what this looks and feels like.

**Comment by Dtphelan1** [comment]
Probably gonna just merge by EOB today if no one else has #thoughts

File: front-end/src/styles/table.scss [hunk 5]
```diff
@@ -1,8 +1,9 @@
 @import '_variables.scss';
 
 .table {
-  height: 85vh;
+  // height: 85vh;
```
Comment (mgramigna) [inline front-end/src/styles/table.scss]: this obsolete?

**Comment by mgramigna** [comment]
Most recent changes look good to me on mobile and web

File: front-end/src/styles/table.scss (hunk 5 above)
Comment (Dtphelan1) [reply front-end/src/styles/table.scss]: lol yes. good catch ;) was experimenting and liked the no-fixed-height better

**Pull request merged**